### DISCUSSION TITLE
does huge logs affects elastalert runs?

### DISCUSSION THREAD

**Discussion by the-arcade-01** [jertel/elastalert2#1066] [Q&A] [answered]
hi, i've around like 50k-60k logs (api call details) that print every minute on my index. On this index i've set an spike alert, to alert me when some api had breach its threshold limit (used filter and query_key correctly with aggregation and metric). When i test this with normal threshold like 100-1000 calls, it gives me correct alert, but when i try load test it with 50k calls it doesn't give me alerts. So, the question is does huge logs affects elastalert runs? Below is the result i get when i click on test button which elastalert-kibana-plugin for each individual alert.

```
<html>
<head» <title> 504 Gateway Time-out</title> </head>
<body>
«center> <h1>504 Gateway Time-out</h1></center>
</body>
</html>
<!- a padding to disable MSIE and Chrome friendly error page -->
<!- a padding to disable MSIE and Chrome friendly error page -->
```

**Comment by jertel** [comment, marked as answer]
Large indices can affect results. You will need to determine if the problem is with the Elasticsearch query performance, or with ElastAlert requesting too much data to be returned. The former will require tuning Elasticsearch or providing more memory, or splitting up the cluster across multiple search nodes. If the problem is that your query is attempting to return 10's of thousands of records, then you need to make sure you are only requesting counts, not actual record data in your ElastAlert 2 rules.

See the [FAQ](https://elastalert2.readthedocs.io/en/latest/recipes/faq.html#i-have-lots-of-documents-and-it-s-really-slow-how-can-i-speed-it-up) for more information.

**Reply by nsano-rururu** [reply]
There is a possibility of Java heap memory exhaustion due to opening too many indexes.
If you cannot increase the CPU and memory, you should consider closing or deleting unnecessary indexes.
